### PR TITLE
Make vendor proxy changes work

### PR DESCRIPTION
**Pull request by snake14**
### Description:

This is fixing the console command to meet the Matomo 5 vendor proxy requirements.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by snake14** [comment]
@sgiehl @bx80 @michalkleiner I found that this plugin didn't have a PR for vendor proxies, so I went ahead and created one. Can one of you please look over my changes and make sure that I got everything?

I tested the command locally and it worked. That's how I caught the switch from Symfony dialog to question. Should that be mentioned in the migration docs somewhere?

**Comment by sgiehl** [comment]
@snake14 I had only created PRs for those plugins that I had installed locally. There might actually more be missing...

Regarding the migration guide... Isn't this part enough:
```
* Directly using any console helpers is now prohibited
  * To ask for user input use the new methods askForConfirmation, askAndValidate or ask
  * For a progress bar use the methods initProgressBar, startProgressBar, advanceProgressBar and finishProgressBar
  * To render tables use the new method renderTable
```

**Comment by snake14** [comment]
Thank you @sgiehl. Sounds good. I'll keep working my way through the list of all of the plugins our team maintains and create any missing PRs.

I guess I just wasn't familiar with the Symfony helper classes, like dialog, so I didn't know what I was looking for. It probably would have helped if that was actually formatted as a bulleted list in the documentation; It would have been easier to spot and pick out that way. I'll see if I can fix that in my open documentation PR.

Edit:
Looks like you already fixed the lists since the last time I refreshed the page. Thank you.

File: Commands/Migrate.php [hunk 1]
```diff
@@ -110,7 +100,7 @@ public function logMessage($message)
         } else {
             $now = Date::now()->getDatetime();
         }
-        $this->output->writeln($message . ' at ' . $now);
+        $this->getOutput()->writeln($message . ' at ' . $now);
```
Comment (AltamashShaikh) [inline Commands/Migrate.php:103]: should we use below ?
```suggestion
        $output->writeln($message . ' at ' . $now);
```

**Review by AltamashShaikh — COMMENTED** [review]
Except 1 change rest all looks good :+1:

File: Commands/Migrate.php (hunk 1 above)
Comment (snake14) [reply Commands/Migrate.php:103]: With Matomo 5, the `output` property of `ConsoleCommand` is private and `$output` isn't being passed into that method. Are you suggesting that I add it as an argument for the `logMessage` method?

Comment (AltamashShaikh) [reply Commands/Migrate.php:103]: sorry, I missed that was a separate function

**Review by AltamashShaikh — APPROVED** [review]

**Pull request merged**